react-carousel/my-app/src/Carousel.tsx

```diff
@@ -0,0 +1,46 @@
+import { useEffect, useState } from 'react';
+import { Image } from './Image';
+import { GrPrevious } from 'react-icons/gr';
+import { GrNext } from 'react-icons/gr';
+import { FaRegCircle } from 'react-icons/fa';
+type Image = {
+  src: string;
+  alt: string;
+};
+type Props = {
+  images: Image[];
+};
+export function Carousel({ images }: Props) {
+  const [index, setIndex] = useState(0);
+  useEffect(() => {
+    const intervalId = setInterval(() => {
```

Comment:
setInterval works fine here. It also works with setTimeout.

The difference is that with setTimeout you get 3 seconds before the next image switches from the time the user clicks. With setInterval, it switches every 3 seconds, regardless. So the user could click and then instantly it rotates to the next image.

Hmm, after thinking about this, I'm probably wrong, since when the index changes the effect will run. Clean-up will clear the interval and the effect will start a new one, first firing in 3 seconds.